doc/hash/hash-design.md

```diff
@@ -689,6 +783,30 @@ module sonic-hash {
         description "First Revision";
     }
 
+    typedef hash-field {
+        description "Represents native hash field";
+        type stypes:hash-field {
```

Comment:
type stypes:hash-field -> type hash-field?

Reply:
@venkatmahalingam it is by design:
```
INNER_DST_IPV4
INNER_DST_IPV6
INNER_SRC_IPV4
INNER_SRC_IPV6
```
These fields are not generic enough to be used for hash.
This is the reason why the list was reduced.

Also, this part is related to original feature implementation - just aligned HLD with current code base